### PR TITLE
Sphinx docs

### PR DESCRIPTION
**Pull request by kaapstorm**
This PR might be for Jenny.

This branch is my exploration into what's involved with including some of the Confluence documentation into the Sphinx documentation. I also took the opportunity to eyeball the Sphinx docs we've got, and wrap the longer lines so that they're easier to read in an editor (which is how I usually read docs in the codebase I'm working with -- a browser is nice, but your IDE is right there -- and reStructuredText is really cool for that, because it's such a readable markup).

So, some feedback on importing docs from Confluence. Here's how I did it:
1. Open Confluence page you want to import.
2. Click Tools, and choose View Source.
3. Right-click, choose "Save Page As...", and save the page as an HTML document locally.
4. In a terminal, convert to reStructuredText with pandoc:
   
   ```
   $ pandoc -f html -t rst my_doc.html > my_doc.rst
   ```

That's it. I found that pandoc gets most things right, but it's rubbish at tables. It just puts each cell in a new paragraph. So tables are time consuming.

Also code literals seem to be converted to single-cell tables. So you need to put in the "::", and remove the block, but that's not too bad.

In conclusion, importing the public documentation from Confluence is going to take more than half an hour, but it should be doable in four to eight (depending on those tables).

Personally, I like the idea of having the documentation with the code -- it's a little easier to keep them up-to-date together -- and code changes that affect docs can be tracked the same way with GitHub. But I'm new, and I'll defer to your opinions. :-)

### CODE REVIEW

**Comment by czue** [comment]
is this still relevant? @kaapstorm

**Comment by kaapstorm** [comment]
Thanks for spotting this @czue! It's has been wallowing on my TODO list for months, and I totally forgot about it. I'm happy for this to be merged. Not sure who the best person to OK it is though. @orangejenny? @esoergel?

**Comment by czue** [comment]
@kaapstorm apologies - I never really looked too closely at this until now. I'm going to make some comments on the PR that probably should have been made a while ago. will let you do with them what you want. I didn't review the content of the mobile docs closely so someone else should also probably take a pass on those (unless it's just copy/pasted from somewhere)

File: docs/common_commcare_mobile_dev_workspace_build_errors.rst [hunk 1]
```diff
@@ -0,0 +1,202 @@
+Common CommCare Mobile Dev Workspace Build Errors
```
Comment (czue) [inline docs/common_commcare_mobile_dev_workspace_build_errors.rst]: It's not clear to my that mobile docs should be in the HQ repo. I think the benefit is that we only need 1 docs site and infratructure around that? The downside is it violates the "keep docs close to code" principle - since this will often be the only reason mobile devs come to edit the HQ repo.

File: docs/technical_overview.rst [hunk 2]
```diff
@@ -0,0 +1,165 @@
+CommCare Technical Overview
```
Comment (czue) [inline docs/technical_overview.rst]: I don't actually think this is a good example of documentation that should live in the repo. This is much more of a public-facing website-content that is more likely to get updated by communications or marketing people that it is to get updated by developers. One criteria that makes sense (to me at least) for what belongs here is - "would I intuitively go edit this thing when I make a change in the code?" This one doesn't seem to fit the bill.

**Comment by czue** [comment]
also, i'm still not on board with line wrapping for the reasons I already mentioned (edits are annoying unless you use an extra tool which somewhat defeats the point of text docs).

**Comment by czue** [comment]
:+1: to the concept, but not sure about the specific examples you've chosen to port over and put into this repo. I would recommend removing all the large content changes and getting this merged in, then we can discuss/move various individual pieces of content in a less blocking way.

**Comment by kaapstorm** [comment]
Splitting this mess into two branches, and PRing separately. It's been **only** a year. Time to get this merged or closed.

**Comment by kaapstorm** [comment]
:tada: Thanks, @snopoke.
